bsip-0074.md

```diff
@@ -11,26 +11,36 @@ Worker: TBD
 
-To ensure the debt positions can always be closed with suffcient smartcoin, it is more feasible to cut off part of the traded collaterals as fee, instead of smartcoin.
+To ensure the debt positions can always be closed with sufficient smartcoin, it is more feasible to charge part of the traded collaterals as the fee, instead of the smartcoin.
 
-As the margin call fee ratio may be obvious higher than market fee, the margin call orders should be placed at the real price without margin call fee to avoid misleading the buyer.
+As the margin call fee ratio may be obviously higher than market fee, the margin call orders should be placed at the actual price without margin call fee. This will avoid misleading the buyer.
 
 # Specification
 Add one new parameter Margin Call Fee Ratio(MCFR) for each smartcoin, which is controlled by the smartcoin owner.
 
-`Margin call order price = settlement price/(MSSR-MCFR)`.
+**Case 1:** In the case of the call order being the maker,
 
-Here settlement price is a new introduced parameter which is defined in BSIP71, settlement price = feed price when there is no bad debt.
+`Margin call match price = settlement price/(MSSR-MCFR)`.
 
-When a margin call trading happens, the buyer sells smartcoin with quantity X and get collaterals in quantity `X*(MSSR-MCFR)/settlement price`, the margin call order owner sells collaterals in quantity `X*MSSR/settlement price` and get smartcoin in quantity X, the delta in paid and received collaterals in `quantity X*MCFR/settlement price` will be paid to the owner of the smartcoin as margin call fee.
+Here `settlement price` is a new parameter introduced as part of [BSIP71](bsip-0071.md). Note that `settlement price = feed price` when there is no bad debt.
+
+When margin call trading happens with the call order as the maker, the buyer sells smartcoin with quantity X and get collaterals in quantity `X*(MSSR-MCFR)/settlement price`, the margin call order owner sells collaterals in quantity `X*MSSR/settlement price` and get smartcoin in quantity X, the delta in paid and received collaterals (quantity `X*MCFR/settlement price`) will be paid to the owner of the smartcoin as the margin call fee.
+
+**Case 2:** In the case of the call order being the taker,
+
+`Margin call match price = settlement price/MSSR`.
+
+Note that MCFR does not come into play for order matching in this case, but it is involved in the calculation of collateral that is returned to the call order owner.
+
+When margin call trading happens with the call order as the taker, the buyer sells smartcoin with quantity X and receives collateral in quantity `X/limit order price`, the margin call order owner sells collateral in quantity `X/limit order price*(1+MCFR)` and gets smartcoin in quantity X, the delta in paid and received collaterals (quantity `X*MCFR/limit order price`) will be paid to the owner of the smartcoin as the margin call fee.
```

Comment:
This is a fair approach, although perhaps not the best.

Also please be aware that `X` is affected by the `TCR` parameter as well as the filling price (but not only the matching price), when the filling price got shifted due to `MCFR`, `X` could be different. This applies to case 1 too.

Reply:
> This is a fair approach, although perhaps not the best.

Would this be better by not reducing the CR of a margin call order that is partially filled?

"When margin call trading happens with the call order as the taker, the buyer sells smartcoin with quantity X and receives collateral in quantity `X*(1-MCFR)/limit order price`, the margin call order owner sells collateral in quantity `X/limit order price` and gets smartcoin in quantity X, the delta in paid and received collaterals (quantity `X*MCFR/limit order price`) will be paid to the owner of the smartcoin as the margin call fee."

Is there something else that would be better?

Reply:
When there is a limit order bidding high, and a call order takes it, it's perhaps "better" (or not) that the call order pays more fees than `X*MCFR/limit_order_price`, e.g. pays `X*MCFR/feed_price`.